draw/src/shader/draw_text.rs

```diff
@@ -661,9 +661,21 @@ impl DrawText {
                         let mut iter = glyph_infos.iter().peekable();
                         while let Some(glyph_info) = iter.next() {
                             let glyph_start = start + glyph_info.cluster;
-                            let glyph_end = start + iter.peek().map_or(string.len(), |glyph_info| glyph_info.cluster);
+
+                            let glyph_end = if let Some(next_info) = iter.peek() {
+                                start + next_info.cluster
+                            } else {
+                                // If it is the last glyph, use the length of the entire string
+                                start + string.len()
+                            };
+                            // Add safety checks.
+                            if glyph_end <= glyph_start || glyph_start >= text.len() {
```

Comment:
This safety check should not be necessary.

The following assumptions should hold about cluster values:
- They are strictly increasing
- They should always be less than or equal to the length of the text

Given these assumptions, since glyph_start is defined as start + glyph_info.cluster, and glyph_end as either start + next_glyph_info.cluster, or the length of the text, then glyph_start < glyph_end should always hold.

Apparently, this is not the case, or you wouldn't have added this check, so one or both of the above assumptions must be broken. That's a bug, and we should fix it accordingly, rather than add special casing to deal with the bug here.